### DISCUSSION TITLE
Test base class

### DISCUSSION THREAD

**Discussion by samcoppock** [wintercms/winter#1146] [General]
In the test documentation

https://wintercms.com/docs/v1.2/docs/plugin/unit-testing#creating-test-cases-and-fixtures

It states that all tests must extend "System\Tests\Bootstrap\TestCase"

however the "modules/system/tests/bootstrap" folder has 3 Test case files which can be extended

TestCase.php
PluginTestCase.php
PluginManagerTestCase.php

the latter two extend the first.

Is it correct that we should always extend 'TestCase'

If yes

What are the "PluginTestCase" and "PluginManagerTestCase" classes for?

Can existing tests continue to use these other 2 classes?

**Comment by bennothommo** [comment]
@samcoppock the docs might need changing there. Pretty much the only one you should use for test cases in plugins is `PluginTestCase`, as this ensures that the plugin is correctly initialised for testing.

The base `TestCase` class can be used for simple tests that do not need to be run in the context of a plugin. The `PluginManagerTestCase` class is, as far as I know, only used for some specific test cases relating to the Plugin Manager itself, and probably should be moved from that directory.
